### PR TITLE
Crate documentation

### PR DESCRIPTION
**Pull request by Jacarte**
Adding README file and documentation for the crate.

### CODE REVIEW

**Comment by Jacarte** [comment]
cc @fitzgen

File: crates/wasm-mutate/README.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+  given the same inputs as the original Wasm module.
+* **determinism:** `wasm-mutate` is deterministic, i.e., given the same input
+  Wasm module and the same seed, it always produces the same mutated
+  output Wasm module.
+* **libfuzzer integration**: `wasm-mutate` integrates well with mutation-based fuzzers like libFuzzer. It
+  reuses the fuzzer's raw input strings. `wasm-mutate` works with the
+  `LLVMFuzzerCustomMutator` hook and the
+  `libfuzzer_sys::fuzz_mutator!` macro.
+  ### Example
+  ```rust
```
Comment (fitzgen) [inline crates/wasm-mutate/README.md:50]:
```suggestion
  `libfuzzer_sys::fuzz_mutator!` macro.
  
  ### Example
  
  ```rust
```

File: crates/wasm-mutate/README.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+  #![no_main]
+
+    use libfuzzer_sys::fuzz_target;
+    use wasmparser::WasmFeatures;
+
+    fuzz_target!(|bytes: &[u8]| {
```
Comment (fitzgen) [inline crates/wasm-mutate/README.md:56]: This example isn't using the `fuzz_mutator!` macro like the bullet point describes.

File: crates/wasm-mutate/README.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+  ```rust
+    let wasmmutate = wasm_mutate::WasmMutate::default()
+            .seed(seed)
+            .fuel(1000)
+            .preserve_semantics(true)
+            .reduce(true);
+
+    while MAX_ITERATIONS > 0 {
+        wasm = wasmmutate.run(&wasm);
```
Comment (fitzgen) [inline crates/wasm-mutate/README.md:119]: Presumably there should be some kind of interestingness test here, or else we could just immediately assign the empty Wasm module to `wasm` and call it a day.

**Review by fitzgen — APPROVED** [review]
Looks good with a couple nitpicks

**Comment by Jacarte** [comment]
Fixed :)

**Comment by Jacarte** [comment]
History squashed since it is not really relevant

**Review by fitzgen — APPROVED** [review]
Thanks!

**Pull request merged**